File: Selenium/Activities/Activity5.py
from selenium import webdriver
from selenium.webdriver.common.by import By
import time


driver = webdriver.Chrome()
driver.maximize_window()
driver.get("https://training-support.net/webelements/dynamic-controls")
print("Title of page is "+driver.title)
checkbox = driver.find_element(By.ID,"checkbox")
if(checkbox.is_displayed()):
    print("checkbox is displayed")
else:
    print("checkbox is not displayed")
toggle_Checkbox = driver.find_element(By.XPATH,"//button[text()='Toggle Checkbox']")
toggle_Checkbox.click()
time.sleep(5)
if(checkbox.is_displayed()):
    print("checkbox is displayed")
else:
    print("checkbox is not displayed")

File: TestNG/Activity5.java
package TestActivity;



import org.openqa.selenium.By;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.chrome.ChromeDriver;
import org.openqa.selenium.support.Color;
import org.testng.Assert;
import org.testng.annotations.AfterClass;
import org.testng.annotations.BeforeClass;
import org.testng.annotations.Test;

public class Activity5 {
	ChromeDriver driver;
	@BeforeClass(alwaysRun = true)
	public  void initiate() {
		driver = new ChromeDriver();
		driver.manage().window().maximize();
		driver.get("https://training-support.net/webelements/target-practice");
	}
	
	@AfterClass(alwaysRun = true)
	public  void close() {
		driver.quit();
	}
	
	@Test
	public void testTitle() {
		Assert.assertEquals(driver.getTitle(), "Selenium: Target Practice");
	}
	
	@Test(groups = {"headerTest"})
	public void testHeader3() {
		WebElement header3 = driver.findElement(By.tagName("h3"));
		Assert.assertEquals(header3.getText(), "Heading #3");
	
		
	}
	@Test(groups = {"headerTest"})
	public void testHeader5() {
		Color header5Color = Color.fromString(driver.findElement(By.cssSelector("h5.text-purple-600")).getCssValue("color"));
        Assert.assertEquals(header5Color.asHex(), "#9333ea");
	}

	@Test(groups = {"buttonTest"})
	public void buttonTest()
	{
		Assert.assertEquals("Emerald", driver.findElement(By.className("emerald")).getText());
	}
	@Test(groups = {"buttonTest"})
	public void buttonTest2() {
		Color button2Color = Color.fromString(driver.findElement(By.xpath("//button[contains(@class, 'purple')]")).getCssValue("color"));
        Assert.assertEquals(button2Color.asHex(), "#581c87");
	}
}
